src/Compilers/CSharp/Portable/Symbols/BaseTypeAnalysis.cs

```diff
@@ -185,6 +184,12 @@ private static (bool definitelyManaged, bool hasGenerics) DependsOnDefinitelyMan
                         continue;
                     }
 
+                    if (field.RefKind != RefKind.None)
+                    {
+                        // A ref struct which has a ref field is never considered unmanaged
+                        return (true, hasGenerics);
```

Comment:
>hasGenerics [](http://example.com/codeflow?start=38&length=11)

Is `hasGenerics` correct at this point, or do we need to look at `field.Type` to determine that?

Reply:
Had the same thought. At a glance this only seems correct if the return type is `(bool definitelyManaged, bool? hasGenerics)`. I haven't looked at the consuming code though. Could be it's correct in the way that you state.

Reply:
This is only ever called from `GetManagedKind` and the semantics are correct. Basically if `isDefinitelyManaged == false` then we never look at `hasGenerics`. To reduce confusion I'm going to move this to a local function inside of `GetManagedKind`.